packages/js/src/helpers/fields/blockEditorSync.js

```diff
@@ -0,0 +1,71 @@
+import { dispatch, select, subscribe } from "@wordpress/data";
+import { debounce, reduce, mapKeys, forEach } from "lodash";
+import { createWatcher, createCollectorFromObject } from "../create-watcher";
+import { STORES, META_FIELDS, SYNC_TIME, POST_META_KEY_PREFIX } from "../../shared-admin/constants";
+import { getPrimaryTerms } from "./primaryTaxonomiesFieldsStore";
+import { transformMetaValue } from "./transform-meta-value";
+
+/**
+ * Creates an updater.
+ * @returns {function} The updater.
+ */
+const createUpdater = () => {
+	const { editPost } = dispatch( STORES.wp.editor );
+	const { getEditedEntityRecord } = select( "core" );
+	const currentPost = select( STORES.wp.editor ).getCurrentPost();
```

Comment:
Calling this already here might lead to a race condition or not updating properly in the site editor